observability/logging.go

```diff
@@ -0,0 +1,222 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package observability
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/google/uuid"
+	"google.golang.org/grpc/binarylog"
+	binlogpb "google.golang.org/grpc/binarylog/grpc_binarylog_v1"
+	iblog "google.golang.org/grpc/internal/binarylog"
+	configpb "google.golang.org/grpc/observability/internal/config"
+	grpclogrecordpb "google.golang.org/grpc/observability/internal/logging"
+)
+
+// translateMetadata translates the metadata from Binary Logging format to
+// its GrpcLogRecord equivalent.
+func translateMetadata(m *binlogpb.Metadata) *grpclogrecordpb.GrpcLogRecord_Metadata {
+	var res grpclogrecordpb.GrpcLogRecord_Metadata
+	res.Entry = make([]*grpclogrecordpb.GrpcLogRecord_MetadataEntry, len(m.Entry))
+	for i, e := range m.Entry {
+		res.Entry[i] = &grpclogrecordpb.GrpcLogRecord_MetadataEntry{
+			Key:   e.Key,
+			Value: e.Value,
+		}
+	}
+	return &res
+}
+
+type cloudLoggingSink struct {
+	callIDToUUID map[uint64]string
+	lock         sync.RWMutex
+	exporter     genericLoggingExporter
+}
+
+var defaultCloudLoggingSink *cloudLoggingSink
+
+func (cls *cloudLoggingSink) getUUID(callID uint64) string {
+	var (
+		u  string
+		ok bool
+	)
```

Comment:
Nit: I hate `var` blocks in most cases.  Why use 4 lines when 2 will work?  Also zero are needed, if you use `:=` below.

Reply:
Replaced with `:=`.

(Some code in the codebase use this pattern. I thought people like the strictness. Now, I guess they are just unfortunate cases of a variable used in different scopes)